include/fs/fs.h

```diff
@@ -160,6 +161,17 @@ struct fs_file_system_t {
 					struct fs_statvfs *stat);
 };
 
+#define FS_O_READ       0x01
+#define FS_O_WRITE      0x02
+#define FS_O_RDWR       (FS_O_READ | FS_O_WRITE)
+#define FS_O_MODE_MASK  0x03
+
+#define FS_O_CREATE     0x10
+#define FS_O_APPEND     0x20
```

Comment:
I would be a little careful about adding `O_APPEND` support.  In POSIX and littlefs this means atomically seek to the end of the file before each write.  In FatFs it appears to mean "seek to end of file when opened".  If the desired semantics is the former then there is no equivalent FatFs flag and the FatFs wrapper would have to be modified after each seek.  If the desired semantics is a one-time move to the end this flag doesn't do that in littlefs so there is no equivalent flag and the littlefs wrapper would have to be modified to move when opened.

I don't see a test for the precise semantics (create a file with content X, then open it with append, write A, seek to the start, write B, and verify that content is XAB).  I suggest doing this, because it may be that FatFS does the same thing as littlefs (though the code suggests it doesn't.)

Reply:
Added the missing test scenario.

Reply:
I think that the desired outcome should be that writing to file opened for append should append data at the end of file. The test I have provided is very simple one, it just checks if file position changes to expected size after appending already existing file.

Reply:
I think you've missed my point: that is **not** the semantics the flag will have on POSIX or littlefs (as documented).

Assume an application where a file is opened with `FS_O_APPEND`, some material is written, then the application seeks to the start, reads a little, then writes something.  We cannot have a situation where the file content is different based on whether the file system is littlefs or FatFs.

Reply:
OK, I understand now.

Reply:
API meeting 14th July:
- Adopt the POSIX understanding of `O_APPEND`
- Use the native LittleFS APPEND flag
- Modify  the FatFS shim so that it seeks to the end before every write when a file is opened with `O_APPEND`

Reply:
Will fix.

Reply:
Done.

Reply:
Done